sdk/ml/azure-ai-ml/azure/ai/ml/operations/_component_operations.py

```diff
@@ -283,16 +283,14 @@ def create_or_update(
             component = _refine_component(component)
         if version is not None:
             component.version = version
-        if not component.version and self._registry_name:
-            # version is required only when create into registry as
-            # we have _auto_increment_version for workspace component.
-            msg = "Component version is required for create_or_update."
-            raise ValidationException(
-                message=msg,
-                no_personal_data_message=msg,
-                target=ErrorTarget.COMPONENT,
-                error_category=ErrorCategory.USER_ERROR,
-            )
+        if not component.version and component._auto_increment_version:
```

Comment:
Can `component.version == 0` ? 
If so then `if not component.version` will resolve true by accident